website/docs/typescript-support.md

```diff
@@ -5,37 +5,22 @@ title: TypeScript Support
 
 ## Setup
 
-Docusaurus supports writing and using TypeScript theme components. To start using TypeScript, add `@docusaurus/module-type-aliases` to your project:
+Docusaurus supports writing and using TypeScript theme components. To start using TypeScript, add `@docusaurus/module-type-aliases` and some `@types` dependencies to your project:
 
 ```bash
-npm install --save-dev typescript @docusaurus/module-type-aliases
+npm install --save-dev typescript @docusaurus/module-type-aliases @types/react @types/react-router-dom @types/react-helmet
 ```
 
-Then add `tsconfig.json` to your project root with following content:
+Then add `tsconfig.json` to your project root with the following content:
 
 ```json title="tsconfig.json"
 {
-  "compilerOptions": {
-    "allowJs": true,
-    "esModuleInterop": true,
-    "jsx": "react",
-    "lib": ["DOM"],
-    "noEmit": true,
-    "noImplicitAny": false
-  },
+  "extends": "@tsconfig/docusaurus/tsconfig.json",
   "include": ["src/"]
 }
 ```
 
-Docusaurus doesn't use this `tsconfig.json` to compile your TypeScript. It is added just for a nicer Editor experience, although you can choose to run `tsc --noEmit` to type check your code for yourself.
-
-Then add `types.d.ts` in your `src` folder with the following content:
-
-```ts title="src/types.d.ts"
-/// <reference types="@docusaurus/module-type-aliases" />
-```
-
```

Comment:
Removed with https://github.com/tsconfig/bases/pull/28/files

This has one side-effect of encouraging people to be explicit in their global types, so for example if they wanted to add Jest tests, it would need to look like:

```
{
  "extends": "@tsconfig/docusaurus",
  "types": ["node",  "jest", "@docusaurus/module-type-aliases"],
  "include": ["src"]
}
```

But I imagine that this is the minority of folks, so I've left it off